dotCMS/src/integration-test/java/com/dotmarketing/factories/MultiTreeAPITest.java

```diff
@@ -0,0 +1,380 @@
+package com.dotmarketing.factories;
+
+import static com.dotcms.util.CollectionsUtils.list;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import com.dotcms.IntegrationTestBase;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.beans.MultiTree;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.common.db.DotConnect;
+import com.dotmarketing.portlets.containers.model.Container;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.htmlpageasset.model.HTMLPageAsset;
+import com.dotmarketing.startup.runonce.Task04315UpdateMultiTreePK;
+import com.google.common.collect.Table;
+
+public class MultiTreeAPITest extends IntegrationTestBase {
+    
+
+    private static final String CONTAINER = "CONTAINER";
+    private static final String PAGE = "PAGE";
+    private static final String CONTENTLET = "CONTENTLET";
+    private static final String RELATION_TYPE = "RELATION_TYPE";
+
+    final static int runs =2;
+    final static int contentlets =5;
+    @BeforeClass
+    public static void initData() throws Exception {
+        IntegrationTestInitService.getInstance().init();
+      //  testUpgradeTask();
+        buildInitalData();
+    }
+    
+    public static void testUpgradeTask() throws Exception {
+        Task04315UpdateMultiTreePK task = Task04315UpdateMultiTreePK.class.newInstance();
+        task.executeUpgrade();
+    }
+    
+    
+    public static void buildInitalData() throws Exception {
+        for(int i=0;i<runs;i++) {
+            for(int j=0;j<contentlets;j++) {
+                MultiTree mt = new MultiTree()
+                        .setContainer(CONTAINER +i)
+                        .setHtmlPage(PAGE)
+                        .setContentlet(CONTENTLET + j)
+                        .setTreeOrder(j)
+                        .setInstanceId(RELATION_TYPE + i);
+
+                APILocator.getMultiTreeAPI().saveMultiTree(mt);
+            }
+        }
+
+    }
+    
+    @Test
+    public  void testDeletes() throws Exception {
+        deleteInitialData();
+        buildInitalData() ;
+        List<MultiTree> all = APILocator.getMultiTreeAPI().getAllMultiTrees();
+        
+        List<MultiTree> list = APILocator.getMultiTreeAPI().getMultiTrees(PAGE);
+
+        deleteInitialData();
+        assertTrue("multiTree deletes", APILocator.getMultiTreeAPI().getAllMultiTrees().size() < all.size() );
```

Comment:
`APILocator.getMultiTreeAPI().getAllMultiTrees()` can be saved in a variable to avoid multiple calls to the same method